submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoMetaQueryServiceTest.kt

```diff
@@ -0,0 +1,123 @@
+import ac.uk.ebi.biostd.persistence.doc.model.DocAttribute
+import ac.uk.ebi.biostd.persistence.doc.model.DocProcessingStatus
+import ac.uk.ebi.biostd.persistence.doc.model.DocSection
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmission
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionMethod
+import ebi.ac.uk.model.constants.SubFields
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.DynamicPropertyRegistry
+import org.springframework.test.context.DynamicPropertySource
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.testcontainers.containers.MongoDBContainer
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import org.testcontainers.utility.DockerImageName
+import java.time.Instant
+
+@ExtendWith(SpringExtension::class)
+@Testcontainers
+@SpringBootTest(classes = [MongoDbReposConfig::class])
+internal class SubmissionMongoMetaQueryServiceTest {
+
+    @Autowired
+    lateinit var testInstance: SubmissionMongoMetaQueryService
+
+    @Autowired
+    lateinit var submissionMongoRepository: SubmissionMongoRepository
+
+    @Test
+    fun getBasicProject() {
+        submissionMongoRepository.save(testDocSubmission(
+            accNo = "accNo1",
+            version = 1,
+            status = DocProcessingStatus.PROCESSED,
+            attributes = listOf(DocAttribute(SubFields.ACC_NO_TEMPLATE.value, "template"))
+        ))
+
+        val result = testInstance.getBasicProject("accNo1")
+
+        assertThat(result.accNo).isEqualTo("accNo1")
+    }
+
+    @Test
+    fun findLatestBasicByAccNo() {
+        submissionMongoRepository.save(testDocSubmission("accNo2", 1, DocProcessingStatus.PROCESSED))
+        submissionMongoRepository.save(testDocSubmission("accNo2", -2, DocProcessingStatus.PROCESSED))
+        submissionMongoRepository.save(testDocSubmission("accNo2", 4, DocProcessingStatus.PROCESSED))
+
+        val lastVersion = testInstance.findLatestBasicByAccNo("accNo2")
+
+        assertThat(lastVersion).isNotNull()
+        assertThat(lastVersion!!.version).isEqualTo(4)
+    }
+
+    @Test
+    fun getAccessTags() {
+    }
+
+    @Test
+    fun `exists by AccNo when exists`() {
+        submissionMongoRepository.save(testDocSubmission("accNo3", 1, DocProcessingStatus.PROCESSED))
+
+        assertThat(submissionMongoRepository.existsByAccNo("accNo3")).isTrue()
+    }
+
+    @Test
+    fun `exist by AccNo when don't exists`() {
+        submissionMongoRepository.save(testDocSubmission("accNo4", 1, DocProcessingStatus.PROCESSED))
+
+        assertThat(submissionMongoRepository.existsByAccNo("accNo5")).isFalse()
+    }
+
+    private fun testDocSubmission(
```

Comment:
It might be a good idea to have this in a factory so it can be reused in other tests like

https://github.com/EBIBioStudies/biostudies-backend-services/blob/master/submission/persistence-common/src/test/kotlin/ac/uk/ebi/biostd/persistence/common/filesystem/ExtSubmissionFactory.kt#L14